store/src/main/java/org/apache/rocketmq/store/queue/RocksDBConsumeQueueStore.java

```diff
@@ -0,0 +1,1296 @@
+
+        this.storePath = StorePathConfigHelper.getStorePathConsumeQueue(messageStoreConfig.getStorePathRootDir());
+        this.rocksDBStorage = new ConsumeQueueRocksDBStorage(messageStore, storePath, 4);
+        this.flushOptions = new FlushOptions();
+        this.flushOptions.setWaitForFlush(true);
+        this.flushOptions.setAllowWriteStall(false);
+
+        this.writeBatch = new WriteBatch();
+        this.bufferDRList = new ArrayList(BATCH_SIZE);
+        this.cqBBPairList = new ArrayList(BATCH_SIZE);
+        this.offsetBBPairList = new ArrayList(BATCH_SIZE);
+        for (int i = 0; i < BATCH_SIZE; i++) {
+            ByteBuffer bbKey = ByteBuffer.allocateDirect(MAX_KEY_LEN);
+            ByteBuffer bbValue = ByteBuffer.allocateDirect(MAX_VALUE_LEN);
+            this.cqBBPairList.add(new Pair(bbKey, bbValue));
+
+            ByteBuffer offsetKey = ByteBuffer.allocateDirect(MAX_KEY_LEN);
+            ByteBuffer offsetValue = ByteBuffer.allocateDirect(17);
+            this.offsetBBPairList.add(new Pair(offsetKey, offsetValue));
+        }
+        this.tempTopicQueueMaxOffsetMap = new HashMap<>();
+        this.maxPhyOffsetBB = ByteBuffer.allocateDirect(8);
+
+        this.topicQueueMinCqOffset = new ConcurrentHashMap(1024);
+        this.topicQueueMaxCqOffset = new ConcurrentHashMap(1024);
+
+        this.consumeQueueCache = new FixedSizeCache(1000000);
+
+        this.scheduledExecutorService =
+            Executors.newSingleThreadScheduledExecutor(new ThreadFactoryImpl("RocksDBConsumeQueueStoreScheduledThread", messageStore.getBrokerIdentity()));
+    }
+
+    @Override
+    public void start() {
+        log.info("RocksDB ConsumeQueueStore start!");
+        this.scheduledExecutorService.scheduleAtFixedRate(() -> {
+            rocksDBStorage.statRocksdb(ROCKSDB_LOG);
+        }, 10, this.messageStoreConfig.getStatConsumeQueueRocksDbIntervalSec(), TimeUnit.SECONDS);
+
+        this.scheduledExecutorService.scheduleWithFixedDelay(() -> {
+            cleanTopic(messageStore.getTopicConfigs().keySet());
+        }, 10, this.messageStoreConfig.getCleanDirtyConsumeQueueIntervalMin(), TimeUnit.MINUTES);
+    }
+
+    @Override
+    public boolean load() {
+        boolean result = this.rocksDBStorage.start();
+        log.info("load rocksdb consume queue {}.", result ? "OK" : "Failed");
+        return result;
+    }
+
+    @Override
+    public void recover() {
+        // ignored
+    }
+
+    @Override
+    public boolean recoverConcurrently() {
+        return true;
+    }
+
+    @Override
+    public boolean shutdown() {
+        this.scheduledExecutorService.shutdown();
+        return shutdownInner();
+    }
+
+    private boolean shutdownInner() {
+        this.consumeQueueCache.shutdown();
+        return this.rocksDBStorage.shutdown();
+    }
+
+    @Override
+    public void putMessagePositionInfoWrapper(DispatchRequest request) throws Exception {
+        if (request == null || this.bufferDRList.size() >= BATCH_SIZE) {
```

Comment:
Bulk put is used here, if the number of requests never reaches BATCH_SIZE, does it never put?